addons/product/models/product_attribute.py

```diff
@@ -526,6 +533,31 @@ def _is_from_single_value_line(self, only_active=True):
             all_values = all_values._only_active()
         return len(all_values) == 1
 
+    def _unarchive_variants(self):
+        """
+        Unarchive all the variants for product_template
+        """
+        variants = self.product_tmpl_id.product_variant_ids.with_context(active_test=False).product_variant_ids
```

Comment:
```suggestion
        variants = self.product_tmpl_id.with_context(active_test=False).product_variant_ids
```
product_variant_ids is repeated